libraries/lib-utility/MemoryStream.h

```diff
@@ -15,20 +15,45 @@
 #include <list>
 #include <vector>
 
+#include "MemoryX.h"
+
 /*!
  * @brief A low overhead memory stream with O(1) append, low heap fragmentation and a linear memory view.
  *
  * wxMemoryBuffer always appends 1Kb to the end of the buffer, causing severe performance issues
- * and significant heap fragmentation. There is no possibility to controll the increment value.
+ * and significant heap fragmentation. There is no possibility to control the increment value.
  *
- * std::vector doubles it's memory size which can be problematic for large projects as well.
+ * std::vector doubles its memory size which can be problematic for large projects as well.
  * Not as bad as wxMemoryBuffer though.
  * 
  */
 class UTILITY_API MemoryStream final
 {
 public:
    using StreamData = std::vector<uint8_t>;
+   using StreamChunk = std::pair<const void*, size_t>;
+
+private:
+   static constexpr size_t ChunkSize =
+      1024 * 1024 - // 1Mb
+      2 * sizeof(void*) - // account for the list node pointers
+      sizeof(size_t); // account for the bytes used member
```

Comment:
Maybe subtract a little more for operator new overhead.  Make a guess what that is.

Reply:
I saw no measurable difference here, to be honest. If CRT decides that it will allocate page size aligned amount of memory - we will have an overhead of 4K (less than 1% percent overhead in this case) and will be likely able to reuse it. If CRT fallbacks to some `mmap` kind of API - there is no overhead that we can control here. Controlling the overhead from std::allocator is more feasible, but I'm not sure if we really need such control. (And the simplest way is to simply use a handwritten allocator)

Anyway, the current bottleneck is very far away from this place both for serialization and deserialization. A better fix would be to perform incremental IO directly from the database, but SQLite3 interface doesn't really have an incremental write API. We can avoid memory "linearization" though because `sqlite3_bind_zeroblob` is a very fast API (not actually committing any pages, rather just setting a propper header)

Reply:
Shrug.  Whatever easy little help we can get.  Yes, measure first.